relay-general/src/store/transactions/processor.rs

```diff
@@ -266,26 +266,34 @@ fn normalize_transaction_name(transaction: &mut Annotated<String>) -> Processing
         .collect::<Vec<_>>();
 
     transaction.apply(|trans, meta| {
+        let mut caps = Vec::new();
         // Collect all the remarks if anything matches.
         for matches in TRANSACTION_NAME_NORMALIZER_REGEX.captures_iter(trans) {
             for name in &capture_names {
                 if let Some(m) = matches.name(name) {
                     let remark =
                         Remark::with_range(RemarkType::Substituted, *name, (m.start(), m.end()));
-                    meta.add_remark(remark);
+                    caps.push((m, remark));
                     break;
                 }
             }
         }
 
-        let changed = TRANSACTION_NAME_NORMALIZER_REGEX
-            .replace_all(trans, "*")
-            .to_string();
-        if *trans != changed && changed != "*" {
+        // Sort by the longest match end.
+        caps.sort_by_key(|(m, _)| m.end());
```

Comment:
nit: Rename `m` to `match` everywhere?

Reply:
`match` is actually the reserved word  😄. But I also can change it to `capture`, since it's basically what it is. What do you think?

Reply:
done in [85a517d](https://github.com/getsentry/relay/pull/1839/commits/85a517d138de733b47ad03ea6dd48d1d7aeddbbd)